tests/config_test.go

```diff
@@ -140,33 +137,63 @@ func (c *config) Check(
 		return c, nil, fmt.Errorf("found secrets")
 	}
 
-	d, f, err := infer.DefaultCheck[config](newInputs)
-	return &d, f, err
+	if v, ok := newInputs["applyDefaults"]; ok && v.IsBool() && v.BoolValue() {
+		d, f, err := infer.DefaultCheck[config](ctx, newInputs)
+		*c = d
+		return &d, f, err
+	}
+
+	// No defaults, so apply manually
+	if v := newInputs["field"]; v.IsString() {
+		c.Field = v.StringValue()
+	}
+	if v := newInputs["not"]; v.IsString() {
+		c.NotSecret = v.StringValue()
+	}
+	if v := newInputs["apply-defaults"]; v.IsBool() {
+		c.ApplyDefaults = v.BoolValue()
+	}
+	return c, nil, nil
 }
 
 func TestInferCustomCheckConfig(t *testing.T) {
 	t.Parallel()
 
-	resp, err := integration.NewServer("test", semver.MustParse("0.0.0"), infer.Provider(infer.Options{
+	s := integration.NewServer("test", semver.MustParse("0.0.0"), infer.Provider(infer.Options{
 		Config: infer.Config[*config](),
-	})).CheckConfig(p.CheckRequest{
-		News: resource.PropertyMap{
-			"field": resource.NewProperty("value"),
-			"nested": resource.NewProperty(resource.PropertyMap{
-				"int":        resource.NewProperty(1.0),
-				"not-nested": resource.NewProperty("not-secret"),
-			}),
-			"not": resource.NewProperty("not-secret"),
-		},
+	}))
+
+	t.Run("with-defaults", func(t *testing.T) {
```

Comment:
really this is "with-DefaultCheck" right? we're applying defaults either way, but in the one case it's manual in the custom Check method and in the other, it's relying on DefaultCheck (with the only actual difference being that DefaultCheck applies secrets).